service/utils.hpp

```diff
@@ -0,0 +1,64 @@
+//
+// Created by milerius on 05/03/19.
+//
+
+#pragma once
+
+#include <string>
+#include <fstream>
+#include "service_strong_types.hpp"
+
+namespace raven
+{
+  inline constexpr const table_ascii_st table_ascii{
+      R"(!"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[]^_`abcdefghijklmnopqrstuvwxyz{|}~')"};
+  inline constexpr const std::size_t table_ascii_len = 94;
+
+  inline std::string random_string(std::size_t random_string_len = 32) noexcept
+  {
+      std::random_device rd;
+      std::mt19937 engine(rd());
+      std::uniform_int_distribution<> dist(0, table_ascii_len);
+      std::string ret;
+      ret.resize(random_string_len);
+      std::generate(begin(ret), end(ret), [&engine, &dist]() { return table_ascii.value()[dist(engine)]; });
```

Comment:
couldn't the lambda be marked `noexcept` ?

Reply:
This is unecessary in this case !

"There are two good reasons for the use of noexcept: First, an exception specifier documents the behaviour of the function. If a function is specified as noexcept, it can be safely used in a non-throwing function, and second, it is an optimisation opportunity for the compiler. noexcept may not call std::unexpectedand may not unwind the stack. The initialisation of a container may cheap move the elements into the container if the move constructor is declared as noexcept. If not declared as noexcept, the elements may be expensive copied into the container."

Here it's not the case, because table_ascii is a constexpr const char*, it's not even a container from the c++ point of view, we will not have any advantage of using noexcept specifier here

Reply:
What about `std::generate` ? The standard library may check if the lambda throws and do optimisations based on that. I think gcc even warns for some specific library functions. I therefor believe it's good style to mark the lambdas properly when used for algorithm calls.